pkg/dockerregistry/server/blobdescriptorservice.go

```diff
@@ -25,6 +38,155 @@ type blobDescriptorService struct {
 	distribution.BlobDescriptorService
 }
 
+// Stat returns a a blob descriptor if the given blob is either linked in repository or is referenced in
+// corresponding image stream. This method is invoked from inside of upstream's linkedBlobStore. It expects
+// a proper repository object to be set on given context by upper openshift middleware wrappers.
 func (bs *blobDescriptorService) Stat(ctx context.Context, dgst digest.Digest) (distribution.Descriptor, error) {
-	return dockerRegistry.BlobStatter().Stat(ctx, dgst)
+	repo, repoFound := RepositoryFrom(ctx)
+	if !repoFound {
+		context.GetLogger(ctx).Errorf("failed to retrieve repository from context")
+	}
+
+	// if there is a repo layer link, return its descriptor
+	desc, err := bs.BlobDescriptorService.Stat(ctx, dgst)
```

Comment:
are we just assuming our `blobDescriptorService{}` was set up to contain a nested `BlobDescriptorService` that does ACL checks?